src/Microsoft.OpenApi/Any/OpenApiAnyCloneHelper.cs

```diff
@@ -32,5 +35,27 @@ public static IOpenApiAny CloneFromCopyConstructor(IOpenApiAny obj)
 
             return obj;
         }
+        
+        /// <summary>
+        /// Clones an instance of <see cref="IOpenApiAny"/> object from the copy constructor
+        /// </summary>
+        /// <param name="obj">The object instance.</param>
+        /// <returns>A clone copy or the object itself.</returns>
+        public static IOpenApiAny CloneFromCopyConstructor<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] T>(T obj) where T : IOpenApiAny
```

Comment:
```suggestion
        public static T CloneFromCopyConstructor<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] T>(T obj) where T : IOpenApiAny
```

Since this is now generic, it might make the API more ergonomic if the result was the same type as what was passed in.